BEP18.md

```diff
@@ -0,0 +1,123 @@
+
+## 2.  Abstract
+
+[State sync](https://docs.binance.org/fullnode.html#state-sync) is a way to help newly-joined users sync the latest status of the binance chain. It syncs the latest sync-able peer's status so that fullnode user (who wants to catch up with chain as soon as possible) doesn't need sync from block height 0 (with a cost that discards all historical blocks locally).
+
+BEP-18 Proposal describes an enhancement of existing state sync implementation to improve user experience. The latest status to be synced is represented by **"snapshot"**, which consists of a manifest file and a bunch of snapshot chunk files. The manifest file summarizes version, height and checksums of snapshot chunk files of this snapshot. The snapshot chunk files encoded essential state data to recover a full node.
+
+This BEP introduces the following details:
+
+- What's the procedure to take a snapshot
+- What's the procedure to sync snapshot from other peers
+- Snapshot (manifest, snapshot chunks) format
+
+## 3.  Status
+
+This BEP is still working in progress.
+
+## 4.  Motivation
+
+We propose this BEP to enhance full node user experience (and ease their pain) on using state sync because of following implementation limitations.
+
+1. Users complain most about state syncing testnet is very slow and usually stuck on some requests. <br> <br> In this enhancement, we want data to respond more evenly across peers so that syncing can continuously make progress and the overall syncing time can reduce from 30 - 45 min to around 5 min.
+
+2. Interruption during state sync (node process get killed because of reboot computer or user impatience) would make already synced data in vain (because the current full node doesn't persist synced part on disk). Worsely it mistakenly writes a lock file prevents user state sync again. <br> <br> In this enhancement, we want support break-resume downloading and keep the consistent status for arbitrarily restart.
+
+## 5.  Specification
+
+State sync will download **manifest** and **snapshot chunks** from other peers.
+
+### 5.1 Take snapshot
+
+There are two ways to take snapshots from a fullnode: automatically or manually. Snapshots will be put under `$HOME/data/snapshot/<height>`. All types involved in the snapshot are encoded by go-amino and compressed by snappy. More details will be explained later.
+
+1. To make fullnode automatically take snapshots, just make sure `state_sync_reactor` in `$HOME/config.toml` is set to true.
+2. To manually take snapshots, stop the node if it is running, then run `./bnbchaind snapshot --home <home> --height <height>`.
+
+If the snapshot taking procedure is interrupted, the node will be still in good status, but it cannot provide the interrupted height for other peers.
+
+Note: automatically take snapshot would keep occupying disk space. Fullnode would not delete them automatically, so the user should periodically delete unneeded snapshots manually if they want to save disk space.
```

Comment:
What prevents you from deleting them automatically?

Reply:
> What prevents you from deleting them automatically?

The frequency of snapshot taken is once per day for binance chain. And in our testnet, it takes 300M per snapshot and should be around just 30M at production considered the account numbers is ten times less than testnet there. Whole year less than 15GB storage cost is not a concern for now.